core/include/bitarithm.h

```diff
@@ -164,6 +164,43 @@ static inline unsigned bitarithm_lsb(unsigned v)
 }
 #endif
 
+/**
+ * @brief   Returns the index of the first set bit in @p state, returns @p state with
+ *          that bit cleared.
+ *
+ * @warning This is a low-level helper function, arguments are not checked.
+ *          It is intended to iterate over a bit map until all bits are cleared.
+ *          Whether it starts with the highest or lowest bit will depend on what is fastest
+ *          on the given hardware.
```

Comment:
Is the result defined when the bit map is all zero? Could you add a warning for this if that is the case?

Reply:
I added a @warning